packages/application-extension/src/index.tsx

```diff
@@ -487,9 +489,17 @@ const serviceWorkerManagerPlugin: JupyterFrontEndPlugin<IServiceWorkerManager> =
   id: '@jupyterlite/application-extension:service-worker-manager',
   autoStart: true,
   provides: IServiceWorkerManager,
-  activate: (app: JupyterFrontEnd) => {
+  optional: [IKernelClient],
+  activate: (app: JupyterFrontEnd, kernelClient?: IKernelClient) => {
     const { contents } = app.serviceManager;
-    return new ServiceWorkerManager({ contents });
+    const serviceWorkerManager = new ServiceWorkerManager({ contents });
+    if (kernelClient !== undefined) {
+      serviceWorkerManager.registerStdinHandler(
```

Comment:
Here registering the stdin handler in the same plugin that creates `ServiceWorkerManager`. Maybe it should be in a separate plugin of its own?

Reply:
I believe that's fine 👍🏽 This logic of `registerStdinHandler` is not used in the case of Shared Array Buffers, right?

Reply:
Correct. The changes for SharedArrayBuffer are in the kernel repo code (xeus and pyodide), in the communciation between the UI thread and web worker. JupyterLite doesn't need to know about that, it just deals with the `input_request` and `input_reply` messages as though they are normal kernel messages.